File: frontend/src/utils/sizeParse.ts
/**
 * Parse human-readable size strings to GB (shared with memoryHint).
 */
export function parseHumanSizeToGb(s: unknown): number | null {
  if (s == null || s === '') return null;
  const str = String(s)
    .trim()
    .toLowerCase()
    .replace(/[,~≈]/g, '')
    .replace(/\s+/g, '');
  const m = str.match(/([\d.]+)\s*(tb|t|gb|g|mb|m)?/);
  if (!m) return null;
  let n = parseFloat(m[1]);
  if (!Number.isFinite(n) || n <= 0) return null;
  const u = m[2] || 'gb';
  if (u === 'tb' || u === 't') n *= 1024;
  else if (u === 'mb' || u === 'm') n /= 1024;
  return n;
}

File: backend/utils/size_parse.py
"""Parse human-readable model size strings to GB (aligned with frontend memoryHint)."""

from __future__ import annotations

import re
from typing import Optional


def parse_human_size_to_gb(value: object) -> Optional[float]:
    if value is None or value == "":
        return None
    text = str(value).strip().lower()
    text = re.sub(r"[,~≈]", "", text)
    text = re.sub(r"\s+", "", text)
    match = re.match(r"([\d.]+)\s*(tb|t|gb|g|mb|m)?", text)
    if not match:
        return None
    try:
        amount = float(match.group(1))
    except ValueError:
        return None
    if not (amount > 0):
        return None
    unit = match.group(2) or "gb"
    if unit in ("tb", "t"):
        amount *= 1024
    elif unit in ("mb", "m"):
        amount /= 1024
    return amount
